app/services/alerts/generate_and_save_alerts_and_benchmarks.rb

```diff
@@ -0,0 +1,81 @@
+module Alerts
+  class GenerateAndSaveAlertsAndBenchmarks
+    def initialize(school:, aggregate_school: nil, benchmark_result_generation_run: nil, framework_adapter: FrameworkAdapter)
+      @school = school
+      @aggregate_school = aggregate_school || AggregateSchoolService.new(school).aggregate_school
+      @benchmark_result_generation_run = benchmark_result_generation_run || BenchmarkResultGenerationRun.create!
+      @framework_adapter = framework_adapter
+    end
+
+    def perform(asof_date = Time.zone.today)
+      ActiveRecord::Base.transaction do
+        @alert_generation_run = AlertGenerationRun.create!(school: @school)
+        @benchmark_result_school_generation_run = BenchmarkResultSchoolGenerationRun.create!(school: @school, benchmark_result_generation_run: @benchmark_result_generation_run)
+
+        relevant_alert_types.each do |alert_type|
+          if alert_type.benchmark == true
+            service = GenerateAlertTypeRunResult.new(school: @school, aggregate_school: @aggregate_school, alert_type: alert_type, use_max_meter_date_if_less_than_asof_date: true)
+            service.benchmark_dates(asof_date).each do |benchmark_date|
+              alert_type_run_result = service.perform(benchmark_date)
+              process_alert_type_run_result(alert_type_run_result)
```

Comment:
This is creating multiple alert run results, but at the moment we just have one per run. So this isn't quite right.

What I'm hoping is that we call analyse on the alert type class once, storing the alert and benchmark data and then if its a benchmark we might call it additional times to generate the benchmark results for those extra dates.

But we need to be clear on what the benchmark dates are.